### PR TITLE
Week 1 assessment tawnchan

### PR DESCRIPTION
**Pull request by tawnchan**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅  ] Coding Challenges
- [✅  ] Coding Concepts
- [ ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

I am not very good with wording, maybe overthinking it.

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -6,63 +6,119 @@
...
+} else {
+    console.log("Something went wrong")
+}
+
 // --------------------1) Create the code that determines if a given number is below boiling point, at boiling point, or above boiling point. Boiling point is 212 degrees Fahrenheit. Use the test variables provided below.
 
-// Describe your process:
+// Describe your process: 
+//create a variable to contain boiling point value.
+const boilingPoint = 212
```
Comment (elysemontano) [inline code-challenges.js:43]: Nice work here!

File: code-challenges.js [hunk 2]
```diff
@@ -6,63 +6,119 @@
...
 
-// const temp = 212
+const temp = 212
 // Expected output: "212 is at boiling point"
 
+if (boilingPoint > temp) {
+    console.log(`${temp} is below boiling point`)
+} else if (boilingPoint < temp) {
+    console.log(`${temp} is above boiling point`)
+} else {
```
Comment (elysemontano) [inline code-challenges.js:61]: Logic looks excellent.  To expand a little deeper, consider using the else as a catch-all for edge cases.  For instance, what if temp is a string?

File: code-challenges.js [hunk 3]
```diff
@@ -6,63 +6,119 @@
...
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
 // Describe your process:
+// Use the .sort to sort the variables in the array from lowest number value to highest number value.
+// use the function and return method to flip the order again so now it returns the highest value to lowest value.
 
 const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
 // Expected output: [82, 80, 79, 77, 76, 73, 72]
 
+sanDiegoSummerTemperatures.sort(function(a, b){return b-a})
```
Comment (elysemontano) [inline code-challenges.js:115]: Awesome work on this!  I would like to see more explanation on what is happening with the sort function and what is being passed into it.

File: coding-concepts.js [hunk 4]
```diff
@@ -7,53 +7,53 @@
...
 
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: it should console.log the values in the array to uppercase.
+// b) Verify and explain: it did not... it returns an error saying .toUpperCase is not a function. I think its because it doesn't work on arrays.
```
Comment (elysemontano) [inline coding-concepts.js:50]: Correct, .toUpperCase is a string method

File: interview-questions.md [hunk 5]
```diff
@@ -8,50 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 1. What is git? What is the difference between git and Github?
 
-Your answer:
+Your answer: Git is a version control software used by developers to keep track of the different version of their work. Github is a platform used by developers to use Git.
 
-Researched answer:
+Researched answer: Git is a free, open-sourced version control system widely used in the programming world. It is a very powerful tool used for sharing and working on code and allows for many people to contribute to the same code base. 
+
+GitHub, Inc., is an Internet hosting service for software development and version control using Git. It provides the distributed version control of Git plus access control, bug tracking, software feature requests, task management, continuous integration, and wikis for every project
```
Comment (elysemontano) [inline interview-questions.md]: Answers must be formed in your own words.

File: interview-questions.md [hunk 6]
```diff
@@ -8,50 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: >, <, >=, <=, !=, !==, ==, ===.
 
-Researched answer:
+Researched answer: i got the same..
```
Comment (elysemontano) [inline interview-questions.md]: A researched answer should expand and explain further on the concept.  In this case, there are several other operators that should be included in here and each one of these should be explained along with their operator name.

File: interview-questions.md [hunk 7]
```diff
@@ -8,50 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Researched answer: Git is a free, open-sourced version control system widely used in the programming world. It is a very powerful tool used for sharing and working on code and allows for many people to contribute to the same code base. 
+
+GitHub, Inc., is an Internet hosting service for software development and version control using Git. It provides the distributed version control of Git plus access control, bug tracking, software feature requests, task management, continuous integration, and wikis for every project
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: >, <, >=, <=, !=, !==, ==, ===.
```
Comment (elysemontano) [inline interview-questions.md:25]: Keep in mind, your answer is supposed to be a reflection of how you would respond if asked this question in an interview.  Using vocabulary and definitions as much as possible will help show your understanding and express interest in the job.

File: interview-questions.md [hunk 8]
```diff
@@ -8,50 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Researched answer:  Arrays are variables that store collections of data in an ordered list. A string is a sequence of one or more characters that may consist of letters, numbers, or symbols.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Proper pair programming consist of a driver(s), person who types out the code and the navigator, the person who steers the direction of the coding process.
+
+good proper pair programming is communicating clearly the problem and objective of the solution and both figuring out how to do so using the available tools such as Github, Zoom and Slack. This way multiple people are working on the project and are able to cross check and validate each other.
```
Comment (elysemontano) [inline interview-questions.md]: Really love this explanation!

File: interview-questions.md [hunk 9]
```diff
@@ -8,50 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Researched answer: Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator,reviews each line of code as it is typed in. The two programmers switch roles frequently.
+
+While reviewing, the observer also considers the "strategic" direction of the work, coming up with ideas for improvements and likely future problems to address. This is intended to free the driver to focus all of their attention on the "tactical" aspects of completing the current task, using the observer as a safety net and guide.
```
Comment (elysemontano) [inline interview-questions.md]: Researched answers must be formed in your own words.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Requesting changes: All researched answers must be formed in your own words.  I see several answers that are copied from sources.  Please change those problems to be worded in your own words and I will approve this assessment.

Other comments: Overall, you did a really great job on this assessment.  Your logic looks good in the code challenges, and it seems like you have a good understanding of the concepts.  Some of your interview answers were really thorough and others were very vague.  Make sure to expand on these questions as much as possible.  This is supposed to mimic an interview setting, and how you would respond to your interviewer.

File: code-challenges.js (hunk 2 above)
Comment (tawnchan) [reply code-challenges.js:61]: Thanks for this.. Ari went this over with me also today.. i understand the idea now.

File: interview-questions.md (hunk 5 above)
Comment (tawnchan) [reply interview-questions.md]: i gave my version of the answer on line 17

File: interview-questions.md (hunk 7 above)
Comment (tawnchan) [reply interview-questions.md:25]: Hi Elyse, i just redid the interview questions and pushed it up into the repo folder. do i do a pull request again? how do i notify you of these changes?  Thank you!

**Review by elysemontano — APPROVED** [review]
Great work on those changes!

**Pull request merged**